CONTRIBUTING.md

```diff
@@ -1,24 +1,430 @@
+
+#### Styling your code
+The [./scripts/style.sh] script makes it easy to style your
+code during development.
+<details>
+<summary>Using the <b>style.sh</b> script</summary>
+
+<!-- The above line is intentionally left blank. -->
+```
+# Usage:
+# ./scripts/style.sh [branch-name | filenames]
+#
+# With no arguments, formats all eligible files in the repo
+# Pass a branch name to format all eligible files changed since that branch
+# Pass a specific file or directory name to format just files found there
+#
+# Commonly
+# ./scripts/style.sh your_branch
+```
+</details>
+
+If your PR is failing CI due to style issues please use
+the style script accordingly. If the style script is not working, ensure you
+have installed the necessary code styling tools outlined in the
+[Getting Started](#getting-started) section.
+
+
+#### Apple development style guides and resources
+
+Refer to the following resources when writing Swift or Objective-C code.
+* Swift
+    * [Google's Swift Style Guide][google-swift-style]
+    * [Swift's API Design Guidelines][swift-api-design-guide]
+* Objective-C
+    * [Google's Objective-C Style Guide][google-objc-style]
+    * [Apple's Coding Guidelines for Cocoa][coding-guidelines-for-cocoa]
+
+#### **An example Git workflow**
+This is a general overview of what the Git workflow may look like, from start to
+finish, when contributing code to Firebase.
+The below snippet is purely for reference purposes and is used to demonstarate
+what the workflow may look like, from start to finish.
+<details>
+<summary>View the workflow</summary>
+
+<!-- The above line is intentionally left blank. -->
+For developers without write access, you'll need to create a fork of Firebase
+instead of a branch. Learn more about forking a repo [here][github-forks].
+```console
+# Update your local master
+git checkout master
+git pull
+
+# Create a development branch
+git checkout -b my_feature_or_bug_fix
+
+# Code, commit, repeat
+git commit -m "a helpful commit message"
+
+# Push your local branch to the remote
+git push --set-upstream origin my_feature_or_bug_fix
+
+# Open a pull request on github.com
+
+# Resolve review feedback on opened PR
+git commit -m "implemented suggestion"
+git push
+
+# Your PR can be merged once it has been approved by a project member 🎉
+```
+</details>
+<br> <!-- This new line is for styling purposes. -->
+
+### Debugging
+Xcode ships with many debugging tools. Learn more about debugging in
+Xcode by watching [WWDC sessions][wwdc-sessions] about debugging and
+viewing the [documentation][xcode-debugging].
+
+### Testing
+Tests are an essential part to building Firebase. Many of the tests
+for Firebase run as part of our continous intengration (CI) setup with
+[GitHub Actions]. _Fixing a bug?_ Add a test to catch potential regressions in
+the future. _Adding a new feature?_ Add tests to test the new or modified APIs.
+
+Oftentimes, tests can be useful in understanding how a particular API works.
```

Comment:
I think it might be a good idea to suggest adding a short snippet how a (new) feature works to the release notes, or add a sample project to the Examples. 

For example, when publishing the latest release notes video (https://www.youtube.com/watch?v=im54lXEujCc&t=62s), we didn't have a good code snippet for Codable in RTDB and didn't have enough time to create a new one from scratch.

Would also be great for consumers of our APIs to see how they're used.

Reply:
Great point, I think that'd be helpful– improving. For future release notes videos, if there aren't any snippets/example projects readily available on the repo, ping me and I can share the corresponding API review document. Those docs should always have samples of how the API is used.